test/role.test.js

```diff
@@ -208,4 +210,64 @@ describe('role model', function() {
 
   });
 
+  describe('listByPrincipalType', function() {
+    var sandbox;
+
+    beforeEach(function() {
+      sandbox = sinon.sandbox.create();
+    });
+
+    afterEach(function() {
+      sandbox.restore();
+    });
+
+    it('should fetch all models assigned to the role', function(done) {
+      var principalTypesToModels = {};
+      var runs = 0;
+      var mappings;
+
+      principalTypesToModels[RoleMapping.USER] = User;
+      principalTypesToModels[RoleMapping.APPLICATION] = Application;
+      principalTypesToModels[RoleMapping.ROLE] = Role;
+
+      mappings = Object.keys(principalTypesToModels);
+
+      mappings.forEach(function(principalType) {
+        var Model = principalTypesToModels[principalType];
+        Model.create({name:'test', email:'x@y.com', password: 'foobar'}, function(err, model) {
+          Role.create({name:'testRole'}, function(err, role) {
+            role.principals.create({principalType: principalType, principalId: model.id}, function(err, p) {
+              var pluralName = Model.pluralModelName.toLowerCase();
+              role[pluralName](function(err, models) {
+                assert(!err);
+                assert.equal(models.length, 1);
+                if (++runs === mappings.length) {
+                  done();
+                }
+              });
+            });
+          });
+        });
+      });
+    });
+
+    it('should apply query', function(done) {
+      User.create({name: 'Raymond', email: 'x@y.com', password: 'foobar'}, function(err, user) {
+        Role.create({name: 'userRole'}, function(err, role) {
+          role.principals.create({principalType: RoleMapping.USER, principalId: user.id}, function(err, p) {
+            var query = {fields:['id', 'name']};
+            sandbox.spy(User, 'find');
```

Comment:
> Haha yes I know this part. :)

Ah ok ha, I thought you were asking what a spy is, but now realize you were talking about the spy function. The function accepts two args (Object, methodString) and creates a spy out of the method associated with the string.

> Agreed... And we aren't doing a good job of this really. But I prefer the same kind of focus you are talking about with PRs so we can avoid having to go off track in the PR comments like this :(.

Sure that makes sense. I didn't want to submit tests with my PR that I weren't comfortable with but I get your point. Since theres no agreed upon module to use for stubs, spies, and mocks...etc, then basically I'm introducing a module thats expected to be used across all tests without any consensus :blush: 

> Since you might not have time, even just starting a conversation about the specifics could be helpful. Just keep in mind that this isn't really a priority.

I like this approach better. I'll take sinon out of this PR and make an issue to discuss choosing a testing module for spies, and mocks...etc

Thanks for the advice.